src/main/java/reposense/model/SupportedDomainUrlMap.java

```diff
@@ -0,0 +1,101 @@
+    private static final String BRANCH_PLACEHOLDER = "BRANCH";
+    private static final String COMMIT_HASH_PLACEHOLDER = "COMMIT_HASH";
+    private static final String FILE_PATH_PLACEHOLDER = "FILE_PATH";
+
+    private static final Map<String, String> GITHUB_MAP = new HashMap<String, String>() {
+        {
+            put(BASE_URL_KEY, "https://github.com/");
+            put(REPO_URL_KEY, "https://github.com/" + ORGANIZATION_PLACEHOLDER + "/" + REPO_NAME_PLACEHOLDER + "/");
+            put(BRANCH_KEY, "tree/" + BRANCH_PLACEHOLDER);
+            put(COMMIT_PATH_KEY, "commit/" + COMMIT_HASH_PLACEHOLDER);
+            put(BLAME_PATH_KEY, "blame/" + BRANCH_PLACEHOLDER + "/" + FILE_PATH_PLACEHOLDER);
+            put(HISTORY_PATH_KEY, "commits/" + BRANCH_KEY + "/" + FILE_PATH_PLACEHOLDER);
+        }
+    };
+    private static final Map<String, String> GITLAB_MAP = new HashMap<String, String>() {
+        {
+            put(BASE_URL_KEY, "https://gitlab.com/");
+            put(REPO_URL_KEY, "https://gitlab.com/" + ORGANIZATION_PLACEHOLDER + "/" + REPO_NAME_PLACEHOLDER + "/");
+            put(BRANCH_KEY, "-/tree/" + BRANCH_PLACEHOLDER);
+            put(COMMIT_PATH_KEY, "-/commit/" + COMMIT_HASH_PLACEHOLDER);
+            put(BLAME_PATH_KEY, "-/blame/" + BRANCH_PLACEHOLDER + "/" + FILE_PATH_PLACEHOLDER);
+            put(HISTORY_PATH_KEY, "-/commits/" + BRANCH_PLACEHOLDER + "/" + FILE_PATH_PLACEHOLDER);
+        }
+    };
+    private static final Map<String, String> BITBUCKET_MAP = new HashMap<String, String>() {
+        {
+            put(BASE_URL_KEY, "https://bitbucket.org/");
+            put(REPO_URL_KEY, "https://bitbucket.org/" + ORGANIZATION_PLACEHOLDER + "/" + REPO_NAME_PLACEHOLDER + "/");
+            put(BRANCH_KEY, "src/" + BRANCH_PLACEHOLDER);
+            put(COMMIT_PATH_KEY, "commits/" + COMMIT_HASH_PLACEHOLDER);
+            put(BLAME_PATH_KEY, "annotate/" + BRANCH_PLACEHOLDER + "/" + FILE_PATH_PLACEHOLDER);
+            put(HISTORY_PATH_KEY, "history-node/HEAD/" + FILE_PATH_PLACEHOLDER + "?at=" + BRANCH_PLACEHOLDER);
+        }
+    };
+    private static final Map<String, String> NOT_SUPPORTED_MAP = new HashMap<String, String>() {
+        {
+            put(BASE_URL_KEY, "UNSUPPORTED");
+            put(REPO_URL_KEY, "UNSUPPORTED");
+            put(BRANCH_KEY, "");
+            put(COMMIT_PATH_KEY, "");
+            put(BLAME_PATH_KEY, "");
+            put(HISTORY_PATH_KEY, "");
+        }
+    };
+
+    private static final SupportedDomainUrlMap DEFAULT_DOMAIN_URL_MAP = new SupportedDomainUrlMap();
+
+    private final Map<String, Map<String, String>> domainUrlMap;
+
+    private SupportedDomainUrlMap() {
+        domainUrlMap = new HashMap<>();
+        domainUrlMap.put("github", GITHUB_MAP);
+        domainUrlMap.put("gitlab", GITLAB_MAP);
+        domainUrlMap.put("bitbucket", BITBUCKET_MAP);
+        domainUrlMap.put(RepoLocation.UNSUPPORTED_DOMAIN_NAME, NOT_SUPPORTED_MAP);
```

Comment:
Do you think it is possible if we only include the mappings if that particular host is used? For instance, if all my repos are on GitHub, the mappings provided to the frontend only contains `GITHUB_MAP`.

Reply:
Oh this is a great point. Yes, it's probably possible to implement. Will add this.